[BUG] Test "parquet_write_test.py::test_hive_timestamp_value[INJECT_OOM]" failed on Databricks

**Describe the bug**
Test "parquet_write_test.py::test_hive_timestamp_value[INJECT_OOM]" failed on Databricks

Note: nightly build/test PASS out of Databricks env, like on Dataproc cluster.


```
 =================================== FAILURES ===================================
 __________________________ test_hive_timestamp_value ___________________________
 [gw1] linux -- Python 3.8.10 /usr/bin/python
 
 spark_tmp_table_factory = <conftest.TmpTableFactory object at 0x7f6f62ab1bb0>
 spark_tmp_path = '/tmp/pyspark_tests//0716-051143-4t97fgdc-10-59-174-113-gw1-2906-1709110285/'
     def test_hive_timestamp_value(spark_tmp_table_factory, spark_tmp_path):
         def create_table(spark, path):
             tmp_table = spark_tmp_table_factory.get()
             spark.sql(f"CREATE TABLE {tmp_table} STORED AS PARQUET " +
             f""" LOCATION '{path}' AS SELECT CAST('2015-01-01 00:00:00' AS TIMESTAMP) as t; """)
         def read_table(spark, path):
             return spark.read.parquet(path)
         data_path = spark_tmp_path + '/PARQUET_DATA'
 >       assert_gpu_and_cpu_writes_are_equal_collect(create_table, read_table, data_path)
 
 ../../src/main/python/parquet_write_test.py:745: 
 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
 ../../src/main/python/asserts.py:267: in assert_gpu_and_cpu_writes_are_equal_collect
     _assert_gpu_and_cpu_writes_are_equal(write_func, read_func, base_path, 'COLLECT', conf=conf)
 ../../src/main/python/asserts.py:242: in _assert_gpu_and_cpu_writes_are_equal
     with_gpu_session(lambda spark : write_func(spark, gpu_path), conf=conf)
 ../../src/main/python/spark_session.py:133: in with_gpu_session
     return with_spark_session(func, conf=copy)
 ../../src/main/python/spark_session.py:100: in with_spark_session
     ret = func(_spark)
 ../../src/main/python/asserts.py:242: in <lambda>
     with_gpu_session(lambda spark : write_func(spark, gpu_path), conf=conf)
../../src/main/python/parquet_write_test.py:738: in create_table
    spark.sql(f"CREATE TABLE {tmp_table} STORED AS PARQUET " +
/databricks/spark/python/pyspark/instrumentation_utils.py:48: in wrapper
    res = func(*args, **kwargs)
/databricks/spark/python/pyspark/sql/session.py:1387: in sql
    return DataFrame(self._jsparkSession.sql(sqlQuery, litArgs), self)
/databricks/spark/python/lib/py4j-0.10.9.5-src.zip/py4j/java_gateway.py:1321: in __call__
    return_value = get_return_value(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

a = ('xro892112', <py4j.clientserver.JavaClient object at 0x7f6f99bf5f70>, 'o64', 'sql')
kw = {}, converted = IllegalArgumentException()

    def deco(*a: Any, **kw: Any) -> Any:
        try:
            return f(*a, **kw)
        except Py4JJavaError as e:
            converted = convert_exception(e.java_exception)
            if not isinstance(converted, UnknownException):
                # Hide where the exception came from that shows a non-Pythonic
                # JVM exception message.
>               raise converted from None
E               pyspark.errors.exceptions.IllegalArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.command.DataWritingCommandExec
E               Execute InsertIntoHadoopFsRelationCommand file:/tmp/pyspark_tests/0716-051143-4t97fgdc-10-59-174-113-gw1-2906-1709110285/PARQUET_DATA/GPU, false, Parquet, [serialization.format=1, mergeschema=false, __hive_compatible_bucketed_table_insertion__=true], Append, `spark_catalog`.`default`.`tmp_table_gw1_2097635381_1`, org.apache.hadoop.hive.ql.io.parquet.serde.ParquetHiveSerDe, org.apache.spark.sql.execution.datasources.InMemoryFileIndex(file:/tmp/pyspark_tests/0716-051143-4t97fgdc-10-59-174-113-gw1-2906-1709110285/PARQUET_DATA/GPU), [t]
E               +- WriteFiles
E                  +- GpuColumnarToRow false
E                     +- GpuProject [1420070400000000 AS t#1615897]
E                        +- GpuRowToColumnar targetsize(104857600)
E                           +- Scan OneRowRelation[]

/databricks/spark/python/pyspark/errors/exceptions.py:234: IllegalArgumentException
----------------------------- Captured stdout call -----------------------------
### CPU RUN ###
### GPU RUN ###
=============================== warnings summary ===============================
../../../../../../databricks/spark/python/pyspark/sql/pandas/utils.py:37: 16 warnings
src/main/python/udf_test.py: 430 warnings
  /databricks/spark/python/pyspark/sql/pandas/utils.py:37: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    if LooseVersion(pandas.__version__) < LooseVersion(minimum_pandas_version):

../../../../../../databricks/spark/python/pyspark/sql/pandas/utils.py:64: 16 warnings
src/main/python/udf_test.py: 296 warnings
  /databricks/spark/python/pyspark/sql/pandas/utils.py:64: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    if LooseVersion(pyarrow.__version__) < LooseVersion(minimum_pyarrow_version):

src/main/python/hash_aggregate_test.py: 32 warnings
  /databricks/spark/python/pyspark/sql/functions.py:723: FutureWarning: Deprecated in 3.2, use sum_distinct instead.
    warnings.warn("Deprecated in 3.2, use sum_distinct instead.", FutureWarning)

src/main/python/hive_delimited_text_test.py: 56 warnings
  /databricks/spark/python/pyspark/sql/catalog.py:763: FutureWarning: createExternalTable is deprecated since Spark 2.2, please use createTable instead.
    warnings.warn(

src/main/python/window_function_test.py::test_lru_cache_datagen[INJECT_OOM]
  /home/ubuntu/spark-rapids/integration_tests/src/main/python/window_function_test.py:1452: UserWarning: Cache info: CacheInfo(hits=11380, misses=971, maxsize=128, currsize=128)
    warnings.warn("Cache info: {}".format(info))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
- generated xml file: /home/ubuntu/spark-rapids/integration_tests/target/run_dir-20230716052256-1MOT/TEST-pytest-1689484976783667901.xml -
=========================== short test summary info ============================
FAILED ../../src/main/python/parquet_write_test.py::test_hive_timestamp_value[INJECT_OOM] - pyspark.errors.exceptions.IllegalArgumentException: Part of the plan is not...
= 1 failed, 18123 passed, 845 skipped, 848 xfailed, 291 xpassed, 847 warnings in 7336.30s (2:02:16) =
 Warning: Permanently added '[34.216.160.175]:2200' (ECDSA) to the list of known hosts.
 receiving incremental file list

```

failed more cases in 321db runtime
```
[2023-07-16T07:13:09.563Z] FAILED ../../src/main/python/parquet_write_test.py::test_hive_timestamp_value[INJECT_OOM] - pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columna...

[2023-07-16T07:13:09.563Z] FAILED ../../src/main/python/schema_evolution_test.py::test_column_add_after_partition[parquet][IGNORE_ORDER({'local': True})] - py4j.protocol.Py4JJavaError: An error occurred while calling o1848911.saveA...

[2023-07-16T07:13:09.564Z] FAILED ../../src/main/python/schema_evolution_test.py::test_column_add_after_partition[orc][IGNORE_ORDER({'local': True})] - py4j.protocol.Py4JJavaError: An error occurred while calling o1849036.saveA...
```

--- a/integration_tests/src/main/python/parquet_write_test.py
+++ b/integration_tests/src/main/python/parquet_write_test.py
@@ -761,8 +761,8 @@ def test_hive_timestamp_value_fallback(spark_tmp_table_factory, spark_tmp_path):
 
     def func_test(create_table, read_table, data_path, conf):
         assert_gpu_fallback_write(
-            lambda spark, path: create_table(spark, path),
-            lambda spark, path: read_table(spark, path),
+            create_table,
+            read_table,
             data_path,
             ['DataWritingCommandExec'],
             conf)

--- a/integration_tests/src/main/python/parquet_write_test.py
+++ b/integration_tests/src/main/python/parquet_write_test.py
@@ -731,6 +731,7 @@ def test_dynamic_partitioned_parquet_write(spark_tmp_table_factory, spark_tmp_pa
     )
 
 # Test to avoid regression on a known bug in Spark. For details please visit https://github.com/NVIDIA/spark-rapids/issues/8693
+@pytest.mark.allow_non_gpu_databricks("DataWritingCommandExec")
 def test_hive_timestamp_value(spark_tmp_table_factory, spark_tmp_path):
 
     def create_table(spark, path):

--- a/integration_tests/src/main/python/parquet_write_test.py
+++ b/integration_tests/src/main/python/parquet_write_test.py
@@ -730,21 +730,44 @@ def test_dynamic_partitioned_parquet_write(spark_tmp_table_factory, spark_tmp_pa
         conf={}
     )
 
-# Test to avoid regression on a known bug in Spark. For details please visit https://github.com/NVIDIA/spark-rapids/issues/8693
-@pytest.mark.allow_non_gpu_databricks("DataWritingCommandExec")
-def test_hive_timestamp_value(spark_tmp_table_factory, spark_tmp_path):
+def hive_timestamp_value(spark_tmp_table_factory, spark_tmp_path, ts_rebase, func):
+    conf={'spark.sql.legacy.parquet.datetimeRebaseModeInWrite': ts_rebase,
+          'spark.sql.legacy.parquet.int96RebaseModeInWrite': ts_rebase}
 
     def create_table(spark, path):
         tmp_table = spark_tmp_table_factory.get()
         spark.sql(f"CREATE TABLE {tmp_table} STORED AS PARQUET " +
-        f""" LOCATION '{path}' AS SELECT CAST('2015-01-01 00:00:00' AS TIMESTAMP) as t; """)
+                  f""" LOCATION '{path}' AS SELECT CAST('2015-01-01 00:00:00' AS TIMESTAMP) as t; """)
 
     def read_table(spark, path):
         return spark.read.parquet(path)
 
     data_path = spark_tmp_path + '/PARQUET_DATA'
-    assert_gpu_and_cpu_writes_are_equal_collect(create_table, read_table, data_path)
-    assert_gpu_and_cpu_are_equal_collect(lambda spark: spark.read.parquet(data_path + '/CPU'))
+
+    func(create_table, read_table, data_path, conf)
+
+# Test to avoid regression on a known bug in Spark. For details please visit https://github.com/NVIDIA/spark-rapids/issues/8693
+def test_hive_timestamp_value(spark_tmp_table_factory, spark_tmp_path):
+
+    def func_test(create_table, read_table, data_path, conf):
+        assert_gpu_and_cpu_writes_are_equal_collect(create_table, read_table, data_path, conf=conf)
+        assert_gpu_and_cpu_are_equal_collect(lambda spark: spark.read.parquet(data_path + '/CPU'))
+
+    hive_timestamp_value(spark_tmp_table_factory, spark_tmp_path, 'CORRECTED', func_test)
+
+# Test to avoid regression on a known bug in Spark. For details please visit https://github.com/NVIDIA/spark-rapids/issues/8693
+@allow_non_gpu('DataWritingCommandExec')
+def test_hive_timestamp_value_fallback(spark_tmp_table_factory, spark_tmp_path):
+
+    def func_test(create_table, read_table, data_path, conf):
+        assert_gpu_fallback_write(
+            lambda spark, path: create_table(spark, path),
+            lambda spark, path: read_table(spark, path),
+            data_path,
+            ['DataWritingCommandExec'],
+            conf)
+
+    hive_timestamp_value(spark_tmp_table_factory, spark_tmp_path, 'LEGACY', func_test)
 
 @ignore_order
 @pytest.mark.skipif(is_before_spark_340(), reason="`spark.sql.optimizer.plannedWrite.enabled` is only supported in Spark 340+")

--- a/integration_tests/src/main/python/parquet_write_test.py
+++ b/integration_tests/src/main/python/parquet_write_test.py
@@ -756,7 +756,7 @@ def test_hive_timestamp_value(spark_tmp_table_factory, spark_tmp_path):
     hive_timestamp_value(spark_tmp_table_factory, spark_tmp_path, 'CORRECTED', func_test)
 
 # Test to avoid regression on a known bug in Spark. For details please visit https://github.com/NVIDIA/spark-rapids/issues/8693
-@allow_non_gpu('DataWritingCommandExec')
+@allow_non_gpu('DataWritingCommandExec', 'WriteFilesExec')
 def test_hive_timestamp_value_fallback(spark_tmp_table_factory, spark_tmp_path):
 
     def func_test(create_table, read_table, data_path, conf):

